cmd/minikube/cmd/start.go

```diff
@@ -514,6 +516,55 @@ func startHost(api libmachine.API, mc cfg.MachineConfig) (*host.Host, bool) {
 	return host, exists
 }
 
+// isInNoProxy checks if ip is included in NO_PROXY env variable.
+func isInNoProxy(ip string) (bool, string) {
+	v := os.Getenv("NO_PROXY")
+
+	if v == "" {
+		return false, ""
+	}
+
+	//  Checking for when provided IP doesn't have CIDIR subnet.
+	//  For example 192.168.39.224
+	if strings.Contains(v, ip) {
+		return true, v
+	}
+
+	// Checking if the ip is included in the CIDIR subnet ranges
+	// For example 192.168.39.15/24
+	noProxyBlocks := strings.Split(v, ",")
+	for _, b := range noProxyBlocks {
+		if yes, _ := isInBlock(ip, b); yes {
+			return true, v
+		}
+	}
+
+	return false, v
+}
+
+// isInBlock checks if ip is a CIDIR block
+func isInBlock(ip string, block string) (bool, error) {
```

Comment:
Go ahead and move proxy functions into it's own proxy package. This will make tests more straight forward, and will be used by https://github.com/kubernetes/minikube/pull/3835/files#diff-8827f86ac03421fa4d9bc2291083aadf